homeassistant/components/serial/sensor.py

```diff
@@ -50,30 +106,76 @@ async def async_setup_platform(hass, config, async_add_entities, discovery_info=
 class SerialSensor(Entity):
     """Representation of a Serial sensor."""
 
-    def __init__(self, name, port, baudrate, value_template):
+    def __init__(
+        self,
+        name,
+        port,
+        baudrate,
+        bytesize,
+        parity,
+        stopbits,
+        xonxoff,
+        rtscts,
+        dsrdtr,
+        value_template,
+    ):
         """Initialize the Serial sensor."""
         self._name = name
         self._state = None
         self._port = port
         self._baudrate = baudrate
+        self._bytesize = bytesize
+        self._parity = parity
+        self._stopbits = stopbits
+        self._xonxoff = xonxoff
+        self._rtscts = rtscts
+        self._dsrdtr = dsrdtr
         self._serial_loop_task = None
         self._template = value_template
         self._attributes = None
 
     async def async_added_to_hass(self):
         """Handle when an entity is about to be added to Home Assistant."""
         self._serial_loop_task = self.hass.loop.create_task(
-            self.serial_read(self._port, self._baudrate)
+            self.serial_read(
+                self._port,
+                self._baudrate,
+                self._bytesize,
+                self._parity,
+                self._stopbits,
+                self._xonxoff,
+                self._rtscts,
+                self._dsrdtr,
+            )
         )
 
-    async def serial_read(self, device, rate, **kwargs):
+    async def serial_read(
+        self,
+        device,
+        baudrate,
+        bytesize,
+        parity,
+        stopbits,
+        xonxoff,
+        rtscts,
+        dsrdtr,
+        **kwargs,
+    ):
         """Read the data from the port."""
         logged_error = False
         while True:
             try:
                 reader, _ = await serial_asyncio.open_serial_connection(
-                    url=device, baudrate=rate, **kwargs
+                    url=device,
+                    baudrate=baudrate,
+                    bytesize=bytesize,
+                    parity=parity,
+                    stopbits=stopbits,
+                    xonxoff=xonxoff,
+                    rtscts=rtscts,
+                    dsrdtr=dsrdtr ** kwargs,
```

Comment:
Looks like a comma is missing.
```suggestion
                    dsrdtr=dsrdtr, **kwargs,
```

Reply:
Sure, I'll just realize that I'haven't committed this last fix.
I can't directly commit the suggestion ( I don't know why )